node/src/accountant/payment_adjuster/non_unit_tests/mod.rs

```diff
@@ -292,140 +251,240 @@ fn make_payable_account(
+    }) / accounts_count as u128;
+    let max_pieces = accounts_count * COEFFICIENT;
+    let number_of_pieces = generate_usize(gn, max_pieces - 2) as u128 + 2;
+    balance_average / COEFFICIENT as u128 * number_of_pieces
+}
+
 fn make_payables_according_to_thresholds_setup(
     gn: &mut ThreadRng,
-    thresholds_to_be_used: &AppliedThresholds,
     accounts_count: usize,
     now: SystemTime,
-) -> Vec<PayableAccount> {
-    match thresholds_to_be_used {
+) -> (Vec<PayableAccount>, AppliedThresholds) {
+    let wallets = prepare_account_wallets(accounts_count);
+
+    let nominated_thresholds = choose_thresholds(gn, &wallets);
+
+    let payables = match &nominated_thresholds {
         AppliedThresholds::Defaulted => make_payables_with_common_thresholds(
             gn,
+            wallets,
             &PRESERVED_TEST_PAYMENT_THRESHOLDS,
-            accounts_count,
             now,
         ),
-        AppliedThresholds::SingleButRandomized { common_thresholds } => {
-            make_payables_with_common_thresholds(gn, common_thresholds, accounts_count, now)
+        AppliedThresholds::CommonButRandomized { common_thresholds } => {
+            make_payables_with_common_thresholds(gn, wallets, common_thresholds, now)
         }
         AppliedThresholds::RandomizedForEachAccount {
             individual_thresholds,
-        } => {
-            let vec_of_thresholds = individual_thresholds
-                .thresholds
-                .as_ref()
-                .left()
-                .expect("should be Vec at this stage");
-            assert_eq!(vec_of_thresholds.len(), accounts_count);
-            make_payables_with_individual_thresholds(gn, vec_of_thresholds, now)
-        }
-    }
-}
+        } => make_payables_with_individual_thresholds(gn, wallets, individual_thresholds, now),
+    };
 
-fn make_payables_with_common_thresholds(
-    gn: &mut ThreadRng,
-    common_thresholds: &PaymentThresholds,
-    accounts_count: usize,
-    now: SystemTime,
-) -> Vec<PayableAccount> {
-    (0..accounts_count)
-        .map(|idx| make_payable_account(idx, common_thresholds, now, gn))
-        .collect::<Vec<_>>()
+    (payables, nominated_thresholds)
 }
 
-fn make_payables_with_individual_thresholds(
-    gn: &mut ThreadRng,
-    individual_thresholds: &[PaymentThresholds],
-    now: SystemTime,
-) -> Vec<PayableAccount> {
-    individual_thresholds
-        .iter()
-        .enumerate()
-        .map(|(idx, thresholds)| make_payable_account(idx, thresholds, now, gn))
+fn prepare_account_wallets(accounts_count: usize) -> Vec<Wallet> {
+    (0..accounts_count)
+        .map(|idx| make_wallet(&format!("wallet{}", idx)))
         .collect()
 }
 
-fn choose_thresholds(gn: &mut ThreadRng, accounts_count: usize) -> AppliedThresholds {
+fn choose_thresholds(gn: &mut ThreadRng, prepared_wallets: &[Wallet]) -> AppliedThresholds {
     let be_defaulted = generate_boolean(gn);
     if be_defaulted {
         AppliedThresholds::Defaulted
```

Comment:
A suggestion from your side: You prefer to use the `return` keyword here, to prevent using the `else` clause. 

I believe you can do the same with line 430, then we don't need to use any `else` clause. 😉